db/query_test.go

```diff
@@ -0,0 +1,706 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package db
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"golang.org/x/net/context"
+)
+
+var sortableKeysResp = map[string]interface{}{
+	"bob":     person{Name: "bob", Age: 20},
+	"alice":   person{Name: "alice", Age: 30},
+	"charlie": person{Name: "charlie", Age: 15},
+	"dave":    person{Name: "dave", Age: 25},
+	"ernie":   person{Name: "ernie"},
+}
+
+var sortableValuesResp = []struct {
+	resp map[string]interface{}
+	want []interface{}
+}{
+	{
+		resp: map[string]interface{}{"k1": 1, "k2": 2, "k3": 3},
```

Comment:
I wish it could do the left-to-right type inference that would let you omit the types from these composite literals, but this seems like the correct way to do it.

Reply:
Ack